game/plugin/skills/herblore/src/herblore.plugin.kts

```diff
@@ -0,0 +1,33 @@
+import org.apollo.game.message.impl.ItemOptionMessage
+import org.apollo.game.message.impl.ItemOnItemMessage
+
+on { ItemOptionMessage::class }
+        .where { option == 1 }
+        .then {
+            val unidentified = Herb[id] ?: return@then
+            it.startAction(IdentifyHerbAction(it, slot, unidentified))
+            terminate()
+        }
+
+on { ItemOnItemMessage::class }
+        .where { (id == PESTLE_MORTAR || id == KNIFE) && (targetId == PESTLE_MORTAR || targetId == KNIFE) }
+        .then {
+            val raw = CrushIngredient[id] ?: return@then
+            it.startAction(CrushIngredientAction(it, raw))
+            terminate()
+        }
+
+on { ItemOnItemMessage::class }
+        . where { id == WATER_VIAL || targetId == WATER_VIAL }
+        . then {
+            val unfinished = UnfinishedPotion[id] ?: return@then
```

Comment:
Same thing with the `on` handler below, you need to make sure what you pass to `UnfinishedPotion` is actually the ingredient id (not the vial of water).